sdks/java/extensions/smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketSink.java

```diff
@@ -272,7 +276,7 @@ public void processElement(ProcessContext c) throws Exception {
 
     @SuppressWarnings("unchecked")
     private ResourceId writeMetadataFile() {
-      final ResourceId tmpFile = fileAssignment.forMetadata();
+      final ResourceId tmpFile = fileAssignment.forMetadata().getMetadata().resourceId();
```

Comment:
`forMetadata().getMetadata()` looks a bit confusing now with 2 different meanings of metadata. We can avoid that by keeping `ResourceId` in our interface and add a `ResourceId -> ReadableFile` fn for reads only.